static/js/components/AssignmentList.jsx

```diff
@@ -0,0 +1,160 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { useSelector, useDispatch } from 'react-redux';
+import * as Actions from '../ducks/source';
+import * as UserActions from '../ducks/users';
+import styles from './AssignmentList.css';
+
+
+function renderAssignment(assignment, deleteAssignment, dispatch, users, observingRunList,
+  instrumentList) {
+  const { requester_id } = assignment;
+  const requester = users[requester_id];
+  const requndef = requester === undefined;
+
+  if (requndef) {
+    dispatch(UserActions.fetchUser(requester_id));
+  }
+
+  const { run_id } = assignment;
+  const run = observingRunList.filter((r) => r.id === run_id)[0];
+  const runundef = run === undefined;
+
+  const instrument_id = !runundef ? run.instrument_id : undefined;
+  const instrument = !runundef ? instrumentList.filter((i) => i.id === instrument_id)[0] :
+    undefined;
+  const instundef = instrument === undefined;
+
+  return (
+    <tr key={assignment.id}>
+      <td>
+        {requndef ? "Loading..." : requester.username}
```

Comment:
You can just do `requester?.name`
Same below with instrument and run
Then the above _undef vars can go